src/components/modal/delete-modal.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import {
+  k8sDeleteResource,
+  K8sModelCommon,
+  K8sResourceCommon,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import {
+  Button,
+  ButtonType,
+  ButtonVariant,
+  Form,
+  ModalVariant,
+  Stack,
+  StackItem,
+  Text,
+  TextContent,
+  TextVariants,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { Formik } from 'formik';
+import { InputField } from '../../shared';
+import { ComponentProps, createModalLauncher } from './createModalLauncher';
+
+type DeleteResourceModalProps = ComponentProps & {
+  obj: K8sResourceCommon;
+  model: K8sModelCommon;
+  description?: string;
+};
+
+export const DeleteResourceModal: React.FC<DeleteResourceModalProps> = ({
+  obj,
+  model,
+  onClose,
+  description,
+}) => {
+  const deleteResource = async () => {
+    await k8sDeleteResource({
+      model,
+      queryOptions: {
+        name: obj.metadata.name,
+        ns: obj.metadata.namespace,
+      },
+    });
+    onClose();
+  };
+
+  return (
+    <Formik onSubmit={deleteResource} initialValues={{ resourceName: '' }} onReset={onClose}>
+      {({ handleSubmit, handleReset, values, isSubmitting, setSubmitting }) => {
+        const input = values.resourceName;
+        const isValid = input === obj.metadata.name;
+        const helpText = !input ? 'missing information' : null;
```

Comment:
The warning should only be shown when input field is touched. Showing a warning as soon as the modal opens up doesn't look good.